[backport] bitcoin-cli: Use symbolic constants

kyuupichan opened the pull request:
This needs to be on top of #337 (Qt Segfault)

Core PRs are bitcoin#8540, bitcoin#9067 and bitcoin#9120

laanwj committed:
Merge #9067: Fix exit codes

4441018 Every main()/exit() should return/use one of EXIT_ codes instead of magic numbers (UdjinM6)
bd0de13 Fix exit codes: - `--help`, `--version` etc should exit with `0` i.e. no error ("not enough args" case should still trigger an error) - error reading config file should exit with `1` (UdjinM6)

laanwj committed:
Merge #9120: bug: Missed one "return false" in recent refactoring in #9067

45d372f Missed one "return false" in recent refactoring in #9067 (UdjinM6)

gandrewstone commented on src/qt/bitcoin.cpp:
```diff
@@ -580,27 +580,28 @@ int main(int argc, char *argv[])
     // User language is set up: pick a data directory
-    Intro::pickDataDirectory();
+    if (!Intro::pickDataDirectory())
+        return EXIT_SUCCESS;
```
why would we exit with a success if we can't get the data directory?

gandrewstone commented on src/qt/bitcoin.cpp:
```diff
@@ -636,7 +637,7 @@ int main(int argc, char *argv[])
     // translated properly.
     if (PaymentServer::ipcSendCommandLine())
-        exit(0);
+        exit(EXIT_SUCCESS);
```
why do we return everywhere else, but exit here?

kyuupichan replied:
Good points; thanks for the careful review Andrew.  I'll dig deeper.

kyuupichan replied:
Agree with both your points; updated appropriately

Commit:
Follow ups to backport of symbolic constants

1) return false consistently with rest of function
2) return EXIT_FAILURE if user cancels datadir choice dialog
